src/app/campaign/campaign.component.ts

```diff
@@ -53,19 +56,21 @@ export interface LocalFlightState {
 })
 export class CampaignComponent implements OnInit, OnDestroy {
   campaignFlights$: Observable<LocalFlightState[]>;
-  campaignSaving: boolean;
+  campaignSaving$: Observable<boolean>;
   routeSub: Subscription;
 
   constructor(
     private route: ActivatedRoute,
     private router: Router,
     private toastr: ToastrService,
     public campaignStoreService: CampaignStoreService,
+    public store: Store<any>,
     private accountService: AccountService,
     private advertiserService: AdvertiserService
   ) {}
 
   ngOnInit() {
+    this.campaignSaving$ = this.store.pipe(select(selectCampaignSaving));
     this.routeSub = this.route.paramMap
       .pipe(
         switchMap((params: ParamMap) => {
```

Comment:
🤔 We'll probably want advertisers, accounts(, and inventories/series) to be NgRx state as well. Advertisers and Accounts are currently RxJS BehaviorSubject stores.

This is the fragile bit with reactive forms where we make sure to load the advertisers and accounts lists ahead of the campaign so the form has the lists to populate the dropdowns and match a value in order to set the form component value. Yuck.

Reply:
I could see that working pretty easily with NgRx yeah.